### PR TITLE
deal with mandatory filter errors as 'templtable'

### LINKED ISSUE #33447: Host pattern warning when templating
- Bug Report
##### ANSIBLE VERSION
```
ansible 2.4.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/u063326/dev/ansible/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Nov 20 2017, 18:23:56) [GCC 5.4.0 20160609]
```

##### CONFIGURATION
No custom config.

##### COMPONENT NAME
core

##### OS / ENVIRONMENT
Ubuntu 16.04

##### SUMMARY
In a playbook, when the hosts are defined with a jinja2 variable some warnings like `Could not match supplied host pattern, ignoring ` appear:

##### STEPS TO REPRODUCE
Create a playbook with variables for hosts.
```
- hosts: "{{ group | mandatory }}"
```
##### EXPECTED RESULTS
No message.

##### ACTUAL RESULTS
Here is the log when the playbook is running:
```
 [WARNING]: Could not match supplied host pattern, ignoring: {{
 [WARNING]: Could not match supplied host pattern, ignoring: group
 [WARNING]: Could not match supplied host pattern, ignoring: |
 [WARNING]: Could not match supplied host pattern, ignoring: mandatory
 [WARNING]: Could not match supplied host pattern, ignoring: }}
```

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
fixes #33447

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

template
##### ANSIBLE VERSION

```
2.4/2.5
```

### CODE REVIEW

File: lib/ansible/template/__init__.py [hunk 1]
```diff
@@ -540,6 +540,8 @@ def is_template(self, data):
                 new = self.do_template(data, fail_on_undefined=True)
             except (AnsibleUndefinedVariable, UndefinedError):
                 return True
+            except AnsibleFilterError as e:
+                return 'Mandatory variable not defined.' == to_text(e)
```
Comment (jborean93) [inline lib/ansible/template/__init__.py:544]: What happens if it is another filter that caused an error and not `mandatory`?

Comment (bcoca) [reply lib/ansible/template/__init__.py:544]: bad things ... need to narrow it down

**Comment by maxamillion** [comment]
needs_revision

**Comment by jimi-c** [comment]
Closing per comments with @bcoca.

**Pull request closed without merging**